Code/GraphMol/Chirality.cpp

```diff
@@ -3774,5 +3774,82 @@ void simplifyEnhancedStereo(ROMol &mol, bool removeAffectedStereoGroups) {
     }
   }
 }
+
+std::vector<std::pair<unsigned int, unsigned int>> findMesoCenters(
+    const ROMol &mol, bool includeIsotopes, bool includeAtomMaps) {
+  std::vector<std::pair<unsigned int, unsigned int>> res;
+  boost::dynamic_bitset<> specifiedChiralAts(mol.getNumAtoms());
+  std::vector<unsigned> ringStereoAts(mol.getNumAtoms(), mol.getNumAtoms());
+  for (const auto atom : mol.atoms()) {
+    atom->clearProp(common_properties::_mesoOtherAtom);
+    if (atom->getChiralTag() > Atom::ChiralType::CHI_UNSPECIFIED) {
+      specifiedChiralAts.set(atom->getIdx(), 1);
+    }
+    int otherIdx = -1;
+    if (atom->getPropIfPresent(common_properties::_ringStereoOtherAtom,
+                               otherIdx) &&
+        otherIdx >= 0) {
+      ringStereoAts[atom->getIdx()] = static_cast<unsigned>(otherIdx);
+    }
+  }
+  // easy case: no atoms with specified chirality
+  if (specifiedChiralAts.none()) {
+    return res;
+  }
+
+  // we will compare the atom ranks with chirality and with only chiral presence
+  // (so that we can distinguish centers with chirality specified and those
+  // without)
+  const bool breakTies = false;
+  const bool includeChiralPresence = true;
+  const bool includeStereoGroups = false;
+  bool includeChirality = true;
+  std::vector<unsigned int> chiralRanks;
+  Canon::rankMolAtoms(mol, chiralRanks, breakTies, includeChirality,
+                      includeIsotopes, includeAtomMaps, includeChiralPresence,
+                      includeStereoGroups);
+  includeChirality = false;
+  std::vector<unsigned int> presenceRanks;
+  Canon::rankMolAtoms(mol, presenceRanks, breakTies, includeChirality,
+                      includeIsotopes, includeAtomMaps, includeChiralPresence,
+                      includeStereoGroups);
+  for (auto i = 0u; i < mol.getNumAtoms(); ++i) {
```

Comment:
```suggestion
  for (auto i = 0u; i < mol.getNumAtoms() - 1; ++i) {
```
to prevent `specifiedChiralAts[j]`, etc from overflowing ?

Reply:
I'm missing something. The extra iteration isn't necessary, but I don't think there's an overflow. The loop over `j` won't do anything if `i=mol.getNumAtoms()-1`
I think adding the `-1` here makes the code a bit harder to read since it forces a "huh? Why is the loop stopping one short?" moment.

Reply:
ok, true. When I read code I keep forgetting that the condition in the for loop is checked before the first iteration.